From cf5783160d260b939ec777f6d2061257cc7a1145 Mon Sep 17 00:00:00 2001
From: Emil Medve <Emilian.Medve@Freescale.com>
Date: Tue, 9 Apr 2013 10:32:28 -0500
Subject: [PATCH 460/547] squash! iommu/fsl: PAMU driver fixes.

Aperture size calculation was off by one byte

Signed-off-by: Emil Medve <Emilian.Medve@Freescale.com>
Change-Id: I10c7d34f5a8c541b8b3d6290629e1d87ea8a2ea0
Reviewed-on: http://git.am.freescale.net:8181/1170
Reviewed-by: Fleming Andrew-AFLEMING <AFLEMING@freescale.com>
Tested-by: Fleming Andrew-AFLEMING <AFLEMING@freescale.com>
[Original patch taken from QorIQ-SDK-V1.4-SOURCE-20130625-yocto.iso]
Signed-off-by: Bin Jiang <bin.jiang@windriver.com>
---
 drivers/iommu/fsl_pamu_domain.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/iommu/fsl_pamu_domain.c b/drivers/iommu/fsl_pamu_domain.c
index a8390ac..776b803 100644
--- a/drivers/iommu/fsl_pamu_domain.c
+++ b/drivers/iommu/fsl_pamu_domain.c
@@ -450,7 +450,7 @@ static int fsl_pamu_domain_init(struct iommu_domain *domain)
 	dma_domain->iommu_domain = domain;
 	/* defaul geometry 64 GB i.e. maximum system address */
 	domain->geometry.aperture_start = 0;
-	domain->geometry.aperture_end = 1ULL << 36;
+	domain->geometry.aperture_end = (1ULL << 36) - 1;
 	domain->geometry.force_aperture = true;
 
 	return 0;
@@ -762,7 +762,7 @@ static  int configure_domain_geometry(struct iommu_domain *domain, void *data)
 	dma_addr_t geom_size;
 	unsigned long flags;
 
-	geom_size = geom_attr->aperture_end - geom_attr->aperture_start;
+	geom_size = geom_attr->aperture_end - geom_attr->aperture_start + 1;
 	/*
 	 * Sanity check the geometry size. Also, we do not support
 	 * DMA outside of the geometry.
-- 
1.8.4.93.g57e4c17

